bin/sozo/src/commands/build.rs

```diff
@@ -155,21 +155,44 @@ impl BuildArgs {
         // directly during the compilation to get the data we need from it.
         config.tokio_handle().block_on(bindgen.generate(None)).expect("Error generating bindings");
 
-        if self.stats != StatOptions::default() {
-            let world = WorldLocal::from_directory(
-                ws.target_dir_profile().to_string(),
-                ws.load_profile_config().unwrap(),
-            )?;
-
-            let world_stat = world.to_stat_item();
-            let mut stats = vec![world_stat];
-
-            for r in world.resources.values() {
-                if r.resource_type() != ResourceType::Namespace {
-                    stats.push(r.to_stat_item());
-                }
+        let world = WorldLocal::from_directory(
+            ws.target_dir_profile().to_string(),
+            ws.load_profile_config().unwrap(),
+        )?;
+        let world_stat = world.to_stat_item();
+        let mut stats = vec![world_stat];
+        for r in world.resources.values() {
+            if r.resource_type() != ResourceType::Namespace {
+                stats.push(r.to_stat_item());
+            }
+        }
+
+        // Check limits and emit warnings for EVERY compilation
+        for stat in &stats {
+            // Sierra file size check (4,089,446 bytes)
+            if stat.sierra_file_size > 4_089_446 {
+                eprintln!(
+                    "{}: Sierra file size ({:.2}MB, {} bytes) exceeds network limit of 4,089,446 \
```

Comment:
Let's rework a bit the text to make it more readable:

1. We can remove warning word, and print in yellow is ok to note a warning.
2. Wdyt about something like
   `Resource <TAG> exceeds network Sierra file size (#bytes)`
   `Resource <TAG> exceeds network Casm byte code size (#felts)`